mitiq/pec/pec.py

```diff
@@ -42,8 +43,9 @@ class LargeSampleWarning(Warning):
 
 def execute_with_pec(
     circuit: QPROGRAM,
-    executor: Callable[[QPROGRAM], float],
-    representations: List[OperationRepresentation],
+    executor: Callable[[QPROGRAM], QuantumResult],
+    observable: Optional[Observable] = None,
+    representations: Optional[List[OperationRepresentation]] = None,
```

Comment:
Yeah, it needs a default argument since observable has a default. What about an empty sequence as default instead of None?

Reply:
Both `None` or  `[ ] `are good for me. 
However if would raise an error (or a warning?) if `representations` is not given, i.e. if it is left to its default value.
This will make clear to the user that one cannot apply PEC without `representations`.

Reply:
Default argument leading to error is a very bad interface. Maybe you would prefer forcing [keyword only arguments](https://www.python.org/dev/peps/pep-3102/)?

```python
def execute_with_pec(
    circuit: QPROGRAM,
    executor: Callable[[QPROGRAM], QuantumResult],
    observable: Optional[Observable] = None,
    *,
    representations: Tuple[OperationRepresentation, ...],
    precision: float = 0.03,
    num_samples: Optional[int] = None,
    force_run_all: bool = True,
    random_state: Optional[Union[int, np.random.RandomState]] = None,
    full_output: bool = False,
) -> Union[float, Tuple[float, Dict[str, Any]]]:
```

```python
pec.execute_with_pec(circuit, executor)
# TypeError: execute_with_pec() missing 1 required keyword-only argument: 'representations'
```

For now I just warned on missing representation(s). I would bring this up for discussion @ Mitiq meeting. The interface of `execute_with_technique` should be uniform for all techniques, so deciding on keyword only arguments is more general than just PEC.